extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -170,12 +183,74 @@ export class Bind {
       dev().fine(TAG, `Initialized ${bindings.length} bindings from ` +
           `${boundElements.length} elements.`);
     });
+    return this.scanPromise_;
+  }
+
+  /**
+   * Removes all bindings nodes with `node` as their parent.
+   *
+   * Returns a promise that resolves after bindings have been removed.
+   *
+   * @param {!Element} node
+   * @return {Promise}
+   *
+   * @private
+   */
+  removeBindingsForNode_(node) {
+    return new Promise(resolve => {
+      // Eliminate bound elements that have node as an ancestor
+      for (let i = this.boundElements_.length - 1; i >= 0; i--) {
+        // TODO(kmh287): We could probably speed this up more
+        const boundElement = this.boundElements_[i].element;
+        if (this.isAncestorOf_(boundElement, node)) {
+          this.boundElements_.splice(i, 1);
+        }
+      }
+
+      // Eliminate elements from the expression to elements map that
+      // have node as an ancestor. Delete expressions that are no longer
+      // bound to elements.
+      const deletedExpressions = [];
+      for (const expression in this.expressionToElements_) {
+        const elements = this.expressionToElements_[expression];
+        for (let j = elements.length - 1; j >= 0; j--) {
+          if (this.isAncestorOf_(elements[j], node)) {
+            elements.splice(j, 1);
+          }
+        }
+        if (elements.length == 0) {
+          deletedExpressions.push(expression);
+          delete this.expressionToElements_[expression];
+        }
+      }
+
+      // Remove the bindings from the evaluator
+      this.evaluator_.removeBindingsForExpressions(deletedExpressions);
+      resolve();
+    });
+  }
+
+  /**
+   * Checks if `potentialParent` is an ancestor node of `node`
+   * @param {!Element} node
+   * @param {!Element} potentialParent
+   * @return {boolean}
+   */
+  isAncestorOf_(node, potentialParent) {
```

Comment:
Nodes have a [_super_ efficient](https://jsperf.com/node-contains-vs-while-loop/6) `#contains` to do this:

```js
isAncestorOf_(node, potentialParent) {
  return potentialParent.contains(node);
}